js/archive.js

```diff
@@ -110,4 +123,13 @@
     var loading = document.getElementById('loading')
     loading.remove()
   }
+
+  function slugify(text) {
+    return text.toString().toLowerCase()
+      .replace(/\s+/g, '-')           // Replace spaces with -
+      .replace(/[^\w\-]+/g, '')       // Remove all non-word chars
+      .replace(/\-\-+/g, '-')         // Replace multiple - with single -
+      .replace(/^-+/, '')             // Trim - from start of text
+      .replace(/-+$/, '');            // Trim - from end of text
+  }
```

Comment:
Slugifying will remove information that we can't get back. We might want to use `encodeURIComponent` here instead. What do you think?

Reply:
I like having clean urls but I guess it doesn't make a ton of sense in this case. Could be cool in the future though to have something like `/shows/:show-id`

Reply:
That seems like a worthwhile consideration, I like the idea of nice looking show urls. 

What if we did a `trim()`, then a light slugify (only `.replace(/\s/g, '-')`, i removed `+`), and then encode? If the user doesn’t want non word chars or two hyphens then they could just edit the input.

Reply:
updated!

Reply:
This works beautifully.